### PR TITLE
Add aria2 to tools list (step 2)

### PR DESCRIPTION
**Pull request by JackBoosY**
1. Add aria2 to the list of tools
2. Add the aria2 path to the cmake parameters.

Requires https://github.com/microsoft/vcpkg/pull/20971 first.

### CODE REVIEW

File: src/vcpkg/build.cpp [hunk 1]
```diff
@@ -752,6 +752,16 @@ namespace vcpkg::Build
             {"_VCPKG_NO_DOWNLOADS", !Util::Enum::to_bool(action.build_options.allow_downloads) ? "1" : "0"},
         };
 
+        if (action.build_options.download_tool == DownloadTool::ARIA2)
+        {
+#if defined(_WIN32) | defined(__APPLE__)
+            variables.push_back({"ARIA2", paths.get_tool_exe(Tools::ARIA2)});
```
Comment (BillyONeal) [inline src/vcpkg/build.cpp:758]: get_tool_exe should be finding the tool on the PATH and always passing it in. For instance, here's what we do for git:

```
billyoneal@Billys-MacBook-Pro vcpkg % ./vcpkg fetch git   
/usr/bin/git
```

Comment (BillyONeal) [reply src/vcpkg/build.cpp:758]: https://github.com/BillyONeal/vcpkg-tool/commit/3fb9e61b217b48b1720a4dde75c982983116e014 may work?

**Comment by BillyONeal** [comment]
Well I didn't mean to push my suggestions into your branch but when I attempted to push to my demo branch it pushed to your branch instead.

Can you take a look at the changes I submitted, and confirm that you agree with them? Otherwise, feel free to chop my edits off (`git push -f`) and do something else. Point is though that we should find the aria2 from the system and always return that rather than potentially sending broken values into cmake.

**Comment by JackBoosY** [comment]
Test result on Linux:
```
usr@usr:/home/vcpkg# ./vcpkg fetch aria2
/usr/bin/aria2c
```
Please note apt only provide aria2 version `1.33.1`, the user may need to compile and install it manually. @BillyONeal

**Comment by BillyONeal** [comment]
Thanks for adding the fun fetch!

**Pull request merged**